spring-integration-file/src/main/java/org/springframework/integration/file/FileWritingMessageHandler.java

```diff
@@ -878,9 +895,12 @@ private static void rename(File source, File target) throws IOException {
 
 		private volatile long lastWrite;
 
+		private volatile long lastFlush;
+
 		FileState(BufferedWriter writer) {
 			this.writer = writer;
 			this.stream = null;
+			this.lastFlush = System.currentTimeMillis();
```

Comment:
`lastFlush` is never updated.
Am I missing something?

Also I wonder if we should update `MessageFlushPredicate.shouldFlush()` for this new `lastFlush` argument as well.